### PR TITLE
fix: No maxLength label is displayed for arrays of items #1701

### PR DESCRIPTION
**Pull request by zalesky**
## What/Why/How?
closes: https://github.com/Redocly/redoc/issues/1701
## Screenshots (optional)
parameters example:
```yaml
parameters:
  - name: some parameter
    in: path
    schema:
      type: array
      items:
        type: string
        maxLength: 80
        contentEncoding: contentEncoding
  - name: some other parameter
    in: path
    schema:
      type: array
      items:
        type: string
        maxLength: 80
        description: description
        pattern: "^[a-f0-9]{8}-[a-f0-9]{4}-4[a-f0-9]{3}-[89aAbB][a-f0-9]{3}-[a-f0-9]{12}$"
  - name: parameter
    in: path
    schema:
      type: string
      maxLength: 80
      description: description
      pattern: "^[a-f0-9]{8}-[a-f0-9]{4}-4[a-f0-9]{3}-[89aAbB][a-f0-9]{3}-[a-f0-9]{12}$"
  - name: diff parameter
    in: query
    schema:
      type: array
      items:
        type: string
        description: My start list description
        minLength: 3
        maxLength: 100
      maxItems: 12
```
Before fix:

Screenshot 2021-10-08 at 13 48 36

After fix:
Screenshot 2021-10-12 at 15 09 16

additional cases:
```yaml
orderId_0:
  type: array
  minItems: 1
  items:
    type: string
    maxLength: 100
    minLength: 1
    enum: [Jr. Test, Dr. Tester]
    pattern: "^[a-f0-9]{8}-[a-f0-9]{4}-4[a-f0-9]{3}-[89aAbB][a-f0-9]{3}-[a-f0-9]{12}$"
    description: "^[a-f0-9]{8}-[a-f0-9]{4}-4[a-f0-9]{3}-[89aAbB][a-f0-9]{3}-[a-f0-9]{12}$"
orderId_1:
  type: string
  maxLength: 80
  description: description
  pattern: "^[a-f0-9]{8}-[a-f0-9]{4}-4[a-f0-9]{3}-[89aAbB][a-f0-9]{3}-[a-f0-9]{12}$"
orderId_2:
  type: array
  minItems: 1
  items:
    type: array
    minItems: 3
    items:
      type: string
orderId_3:
  type: array
  items:
    type: string
orderId_4:
  type: array
  items:
    type: string
    format: url
```
Screenshot 2021-10-12 at 15 10 48

## Check yourself

- [x] Code is linted
- [x] Tested
- [ ] All new/updated code is covered with tests

### CODE REVIEW

**Comment by jeremyfiel** [comment]
Can this be merged? @RomanHotsiy

**Comment by RomanHotsiy** [comment]
@jeremyfiel no, not yet. We're still trying to find an optimal UI for this.

**Comment by jeremyfiel** [comment]
can I suggest using the following text blocks? 

following use cases of `minLength`, `maxLength`, and `pattern` for strings and range (`minimum` and `maximum`), `multipleOf` for integers.  Also, `inclusiveMinimum`, `inclusiveMaximum`, `exclusiveMinimum`, or `exclusiveMaximum` can be extended for the range example.
I think the `pattern` could become very long and messy on the UI, maybe a tooltip over the pattern component might be helpful, just make sure it's accessible by a screen reader.

`Array of strings` _**having**_ `<= 80 characters`
```yaml
      properties:
        name:
          description: 'this is a description'
          type: array
          items:
            type: string
            maxLength: 80
```
`Array of strings` _**having**_ `^[a-z]$`
```yaml
      properties:
        name:
          description: 'this is a description'
          type: array
          items:
            type: string
            pattern: '^[a-z]$'
```
`Array of strings` _**having**_ `>= 2 characters` _**and**_ `<= 80 characters`
```yaml
      properties:
        name:
          description: 'this is a description'
          type: array
          items:
            type: string
            minLength: 2
            maxLength: 80
```
`Array of strings` _**having**_ `>= 2 characters` _**and**_ `<= 80 characters` _**and**_ `^[a-z]$`
```yaml
      properties:
        name:
          description: 'this is a description'
          type: array
          items:
            type: string
            minLength: 2
            maxLength: 80
            pattern: '^[a-z]$'
```
`Array of integers` _**as a**_ `multipleOf 5`
```yaml
      properties:
        name:
          description: 'this is a description'
          type: array
          items:
            type: integer
            multipleOf: 5
```
`Array of integers` _**as a**_ `multipleOf 5` _**with a**_ `maximum of 10`
```yaml
      properties:
        name:
          description: 'this is a description'
          type: array
          items:
            type: integer
            multipleOf: 5
            maximum: 10
```
`Array of integers` _**as a**_ `multipleOf 5` _**with a**_ `minimum of 5` _**and**_ `maximum of 10`
```yaml
      properties:
        name:
          description: 'this is a description'
          type: array
          items:
            type: integer
            multipleOf: 5
            minimum: 5
            maximum: 10
```
`Array of integers` _**between**_ `[1 ... 100]`
```yaml
      properties:
        name:
          description: 'this is a description'
          type: array
          items:
            type: integer
            minimum: 1
            maximum: 100
```
`Array of integers` _**between**_ `[1 ... 99]`
```yaml
      properties:
        name:
          description: 'this is a description'
          type: array
          items:
            type: integer
            minimum: 1
            inclusiveMinimum: true
            exclusiveMaximum: true
            maximum: 100
```

**Comment by Oprysk** [comment]
updated to:
image

File: src/components/Fields/ArrayItemDetails.tsx [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+import { TypeFormat, TypePrefix } from '../../common-elements/fields';
+import { ConstraintsView } from './FieldContstraints';
+import { Pattern } from './Pattern';
+import { SchemaModel } from '../../services';
+import styled from '../../styled-components';
+
+export function ArrayItemDetails({ schema }: { schema: SchemaModel }) {
+  if (!schema || (schema.type === 'string' && !schema.constraints.length)) return null;
+
+  return (
```
Comment (roberfi) [inline src/components/Fields/ArrayItemDetails.tsx:11]: Hi. This is causing that type label of arrays of any type except string is followed by `[ items ]`:
image

**Pull request merged**